WebRoot/public/modern.jsp

```diff
@@ -8,6 +8,14 @@
 %>
 <script TYPE="text/javascript">
     localStorage.setItem("csrfToken" , "${csrfToken}");
-    var url = window.location.origin + "/modern/"
-    window.location.href = url;
+    
+    const currentURL = new URL(window.location.href);
+    const redirectToPath = currentURL.searchParams.get("appRedirect");
+
+    if (redirectToPath.indexOf('/modern/') === 0) {
```

Comment:
Please put code comments to describe each and every case for future reference.

Reply:
Done.